game-core/src/main/java/games/strategy/triplea/ui/TripleAFrame.java

```diff
@@ -1895,7 +1896,7 @@ public void actionPerformed(final ActionEvent ae) {
           try {
             final File f = TripleAMenuBar.getSaveGameLocation(TripleAFrame.this);
             if (f != null) {
-              try (FileOutputStream fout = new FileOutputStream(f)) {
+              try (final FileOutputStream fout = new FileOutputStream(f)) {
```

Comment:
This final keyword is redundant, because variables in try-with-resource statements must be final anyways.

Reply:
Indeed, does it cause a warning in eclipse? 

It's spotlessApply that added this. I recommend we ignore this, it's been a few times this gets pointed out and it's not much benefit IMO to omit the unnecessary keyword when we include it everywhere else.

/cc @ssoloff

Reply:
> It's spotlessApply that added this

I just ran `spotlessApply` on `master` with the identical changes in this file (sans the `final`), and it does not add the `final` modifier.  Possibly you ran something in your IDE around the same time.

As @RoiEXLab pointed out, we removed the `final` modifiers in try-with-resources blocks some time ago.  If `spotlessApply` were to add them, we'd have about a hundred violations, thus causing the build to fail.

Reply:
I thought `spotlessApply` was being smart enough to only look at modified lines, no, it turns out it was save-actions after-all modifying the file in the background.

Reply:
@RoiEXLab , @ssoloff what are your thoughts? Can we move forward? How can this PR be unblocked?

Reply:
It appears https://github.com/dubreuia/intellij-plugin-save-actions/issues/168 will be available in the next release.

Reply:
Yay!  Let's go with option (B) then as it is a short-term situation.

Reply:
- Redundant final modifier has been deleted: 8d2f84358, resolving this convo :smile: 
- Checkstyle rule to enforce this as provided by @ssoloff has been re-opened: https://github.com/triplea-game/triplea/pull/4628  :+1: